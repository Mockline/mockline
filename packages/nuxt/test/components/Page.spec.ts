import { describe, it, expect } from 'vitest'
import type {TypeOf} from "zod";
import ComponentRender from "../component-render";
// @ts-ignore
import { MPage } from "#components";

describe('Page', () => {
  it.each([
    ['base case', {}],
    ['wrapper', { props: { wrapper: 'relative' } }],
    ['left', { props: { left: 'relative' } }],
    ['right', { props: { right: 'relative' } }],
    ['<MPage><template #left>left</template><template #default>default</template><template #right>right</template></MPage>>']
    // @ts-expect-error
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof MPage.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MPage)
    expect(html).toMatchSnapshot()
  })
})
