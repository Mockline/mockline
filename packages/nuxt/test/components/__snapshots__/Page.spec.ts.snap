// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page > renders <MPage><template #left>left</template><template #default>default</template><template #right>right</template></MPage>> correctly 1`] = `
"<div class="flex sm:flex-row flex-col">
  <div class="flex flex-col">left</div>
  <div class="flex flex-col gap-4 w-full">default</div>
  <div class="flex flex-col">right</div>
</div>
&gt;"
`;

exports[`Page > renders base case correctly 1`] = `
"<div class="flex sm:flex-row flex-col">
  <!--v-if-->
  <div class="flex flex-col gap-4 w-full"></div>
  <!--v-if-->
</div>"
`;

exports[`Page > renders left correctly 1`] = `
"<div class="flex sm:flex-row flex-col">
  <!--v-if-->
  <div class="flex flex-col gap-4 w-full"></div>
  <!--v-if-->
</div>"
`;

exports[`Page > renders right correctly 1`] = `
"<div class="flex sm:flex-row flex-col">
  <!--v-if-->
  <div class="flex flex-col gap-4 w-full"></div>
  <!--v-if-->
</div>"
`;

exports[`Page > renders wrapper correctly 1`] = `
"<div class="flex sm:flex-row flex-col relative">
  <!--v-if-->
  <div class="flex flex-col gap-4 w-full"></div>
  <!--v-if-->
</div>"
`;
